scantools/run_rendering.py

```diff
@@ -26,17 +26,23 @@ def run(capture: Capture, session_id: str, mesh_id: str = 'mesh',
     depths = session.depths
     if depths is None:
         depths = session.depths = RecordsDepth()
-    output_dir = capture.data_path(session_id)
+    output_dir = capture.proc_path(session_id)
     prefix = 'render'
     for ts, camera_id in tqdm(session.images.key_pairs()):
-        pose_cam2w = session.trajectories[ts, camera_id]
+        if session.rigs is None:
+            pose_cam2w = session.trajectories[ts, camera_id]
+        else:
+            rig_id = next(iter(session.trajectories[ts]))
+            rig_from_cam = session.rigs[rig_id, camera_id]
+            world_from_rig = session.trajectories[ts, rig_id]
+            pose_cam2w = world_from_rig * rig_from_cam
+
         camera = session.sensors[camera_id]
         rgb, depth_map = renderer.render_from_capture(pose_cam2w, camera)
 
         image_path = session.images[ts, camera_id]
-        depth_path = Path(prefix, image_path).as_posix() + '.depth.png'
-        render_id = f'{prefix}/{camera_id}'
```

Comment:
I am not sure why this was needed, maybe @sarlinpe remembers

Reply:
The changes on line 37 removes the `jpg` from the filname: ...jpg.depth.png (there is not problem there with this changes).

The changes on line 38 removes the `render/` prefix in the device_id: `render/cam-0`, which doesn't exists.

Reply:
I guess that the prefix meant to distinguish sensor vs rendered depth. In the end we ran `run_rendering` only on NavVis data, where this there is no sensor depth, so this doesn't matter much. You might need to update `lift_points_to_3d` then: https://github.com/microsoft/lamar-benchmark/blob/147c9a4b40c49f9f6e48854996fc20f4d531d9c6/scantools/proc/alignment/scan.py#L49

Reply:
All the usages of `lift_points_to_3d` in the code doesn't explicitly set `depth_prefix`, which means changing `depth_prefix: Optional[str] = 'render'` to `depth_prefix: Optional[str] = None` should do it (to be consistent with the current changes of this PR).

Although, the old datasets would need to be re-converted. So, the options are:
1. Perhaps adding an argument with a default "render":
`parser.add_argument('--depth_prefix', default='render'` in the `run_rendering.py` script. Then, the default output of the script would be exactly as now, but I also can modify the behavior, setting --depth_prefix to empty, the to get what I can (a valid device_id for the depth maps).
2. Add "`render/device_id`" into `sensor.txt` (duplicating the entries). Even though the old datasets won't be valid in this sense, everything still will work because of the `depth_prefix` in `lift_points_to_3d`.
3. `depth_prefix: Optional[str] = None`
4. Not changing this and adding a warning in the CAPTURE.md, that "render/" prefix should be ignored.

Probably option 2. is the best compromise. What do you think?

Reply:
I don't like option 2 because conceptually these renders indeed correspond to `device_id`. How about going for option 3 and updating `lift_points_to_3d` with the following:
```diff
+    if depth_prefix is None and sensor_id not in session.depths[ts]:
+        depth_prefix = 'render'  # Legacy data format.
     if depth_prefix is not None:
         sensor_id = depth_prefix + '/' + sensor_id
     depth = read_depth(data_path / session.depths[ts, sensor_id])
```
I think that this should not break existing datasets.